### PR TITLE
Updated the code

### PR DESCRIPTION
**Pull request by rTreutlein**
works with the newest version of SpockBot and OpenCog

### CODE REVIEW

File: opencog/opencog/spacetime/atom_types.script [hunk 1]
```diff
@@ -15,6 +15,7 @@ IMAGINARY_STRUCTURE_NODE <- OBJECT_NODE
 TIME_NODE <- NODE
 TIME_DOMAIN_NODE <- NODE
 AT_TIME_LINK <- ORDERED_LINK
+AT_LOCATION_LINK <- ORDERED_LINK
```
Comment (linas) [inline opencog/opencog/spacetime/atom_types.script]: Im'm confused. This suggests that a large portion of opencog has been copied into this repo? This should be fixed ...

**Comment by linas** [comment]
OK< to merge, as far as I'm concerned. Anyone else want to review this?

**Comment by ngeiswei** [comment]
Personally I'd rather not to [review it], unless I really have to. Who else is using that repo?

File: opencog/opencog/spacetime/atom_types.script (hunk 1 above)
Comment (linas) [inline opencog/opencog/spacetime/atom_types.script]: never mind, I just looked its not as much as I thought. @ngeiswei would you care to review the 3D code and git-pull it into the main opencog repo? or should I?

**Comment by linas** [comment]
OK, done, I'll try to merge the spacemap/spatial code too.

File: opencog/opencog/spacetime/atom_types.script (hunk 1 above)
Comment (rTreutlein) [inline opencog/opencog/spacetime/atom_types.script]: Yea that is why i created the Issues about merging that stuff into the opencog repo.

**Comment by AmeBel** [comment]
fyi, roman has merge write to this repo.

**Comment by ngeiswei** [comment]
I'm really lost, to merge this to the opencog repo? This isn't even formally forked from opencog (or is it). I need more understanding as to how this code got created and what it's doing. I'm looking at it right now...

**Comment by ngeiswei** [comment]
Oh, it's been merged. Cool, so I'll move on to something else for now.

**Comment by rTreutlein** [comment]
@ngeiswei The opencog folder in this repo currently has to be copy pasted over the opencog repo before building it. And only those changes should be merged with the opencog repo.

**Comment by ngeiswei** [comment]
It would have been better to fork the entire opencog repo, or (if having some standalone component is intended) created a special branch from opencog with the associated changes.

**Comment by linas** [comment]
OK ... so, here's the deal:  The OpenCog repo had a 3DSpaceMap branch. I merged that branch, twice, because I missed a pull request the first time.  The result was some significant changes to the 3DSpaceMap code...

I looked at the diff between what is here, and what is there, and I see more huge changes.  For the moment, I am assuming the code here is newer than the code in opencog.  Starting to review now.

**Comment by linas** [comment]
Or not. I'm going to sleep.

Roman, can I ask a favor? It looks like you are set up to actually test this code (I am not).  Could you tell me if this code works, using the version of spatial in the opencog repo? Or does it really require the code here?  Do you think the code here is newer?

If so I can merge it .. I have a pull request ready to go, here: https://github.com/opencog/opencog/pull/1888 but the above questions need to be resolved, first.

**Comment by rTreutlein** [comment]
@linas Not sure what code is the newest. Ill test it and report back.

**Comment by linas** [comment]
Oh, and just to be clear: I propose that only the 3DSpaceMap code be moved to the opencog directory.  The rest of the stuff here should stay here: viz, spock, world, minecraft_bot stays here.  

Right?  I want to think of this as a kind of "peer" to e.g. the hanson robotics code: its just another ROS thingy that opencog can control; its not an actual part of opencog...

**Comment by rTreutlein** [comment]
Yes just merge the stuff in the opencog folder the rest stays here.

And it seems the code here is required. Not sure if it is the newest but the minecraft_bot won't work with what is in the opencog repository right now.

**Comment by rTreutlein** [comment]
But i don't think the code in here is the newest. Because it still uses the Octree3DMapManager which was removed in this pullrequest:
https://github.com/opencog/opencog/pull/1749

**Pull request merged**